course/page/code.py

```diff
@@ -97,11 +98,19 @@ def debug_print(s):
                 "unix://var/run/docker.sock")
         docker_tls = getattr(settings, "RELATE_DOCKER_TLS_CONFIG",
                 None)
-        docker_cnx = docker.Client(
-                base_url=docker_url,
-                tls=docker_tls,
+
+        if platform.system().lower().startswith("linux"):
+            docker_cnx = docker.Client(
+                    base_url=docker_url,
+                    tls=docker_tls,
+                    timeout=docker_timeout,
+                    version="1.19")
+        else:
+            from docker.utils import kwargs_from_env
+            docker_cnx = docker.Client(
                 timeout=docker_timeout,
-                version="1.19")
+                **kwargs_from_env(assert_hostname=False)
```

Comment:
Thanks! That said, I think it's a risky assumption to say that this'll only be used for development. The simple fact is that this makes it possible to run Relate on Windows and OSX, and we should plan for folks using it that way--and thus make it bulletproof enough to be used in that setting. As such, it should aspire to using reasonable security practices, and `assert_hostname=False` doesn't strike me as one. It can be in there, commented out by default with an explanation, but it shouldn't be the default.